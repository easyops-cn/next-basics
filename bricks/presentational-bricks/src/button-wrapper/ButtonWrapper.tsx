import React from "react";

import styles from "./index.module.css";

export interface ButtonWrapperProps {
  description: string;
  brick: string | any;
  properties: any;
}

export function ButtonWrapper(props: ButtonWrapperProps): React.ReactElement {
  return (
    <div className={styles.buttonWrapper}>
      <p>{props.description}</p>
      <props.brick
        ref={(element: HTMLElement) => {
          if (element) {
            Object.assign(element, props.properties);
          }
        }}
      ></props.brick>
    </div>
  );
}
