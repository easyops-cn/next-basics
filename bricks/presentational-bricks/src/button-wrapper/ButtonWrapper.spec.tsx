import React from "react";
import { shallow } from "enzyme";
import { ButtonWrapper } from "./ButtonWrapper";

describe("ButtonWrapper", () => {
  it("should work", () => {
    const wrapper = shallow(<ButtonWrapper brick="Component" />);
    expect(wrapper).toMatchSnapshot();
  });
});
