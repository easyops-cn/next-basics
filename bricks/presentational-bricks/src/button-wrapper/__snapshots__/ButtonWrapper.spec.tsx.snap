// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWrapper should work 1`] = `
<div
  className="buttonWrapper"
>
  <p />
  <Component />
</div>
`;
